Sources/Rules.swift

```diff
@@ -5171,15 +5171,21 @@ public struct _FormatRules {
         //    and it's more difficult to track the usage of other declaration
         //    types like `init`, `subscript`, `operator`, etc.
         let allowlist = ["let", "var", "func", "typealias"]
+        let overrideList: [DeclarationType] = [.overriddenProperty, .overriddenMethod]
 
         // Collect all of the `private` or `fileprivate` declarations in the file
         var privateDeclarations: [Declaration] = []
         formatter.forEachRecursiveDeclaration { declaration in
             guard allowlist.contains(declaration.keyword),
                   let name = declaration.name,
-                  !(formatter.options.preservedPrivateDeclarations.contains(name))
+                  !(formatter.options.preservedPrivateDeclarations.contains(name)),
+                  !formatter.hasObjcAttribute(declaration)
             else { return }
 
+            // Do not collect any `override` method or property declarations
+            let declarationType = formatter.declarationType(of: declaration.keyword, with: declaration.tokens, allowlist: overrideList)
+            guard !(overrideList.contains(declarationType)) else { return }
```

Comment:
Both `@objc` and `override` are modifiers on the declaration that should be found if you use the existing `modifiersForDeclaration(at:contains:)` helper.

Maybe we can add a helper on `Declaration` that lets you check if the declaration has a given modifier:

```swift
/// Whether or not this declaration has the given modifier or attribute
func hasModifier(_ modifier: String) -> Bool {
    let parser = Formatter(openTokens)

    guard let keywordIndex = parser.index(of: .keyword(keyword), after: 0) else {
        return false
    }

    return parser.modifiersForDeclaration(at: keywordIndex, contains: { $1 == modifier })
}
```

Then that way you can simplify this to:

```suggestion
                  !formatter.hasModifier("override"),
                  !formatter.hasModifier("@objc"),
                  !formatter.hasModifier("@IBOutlet"),
                  !formatter.hasModifier("@IBAction")
            else { return }
```

Reply:
This would be even better:

```swift
/// The modifiers and attributes preceeding this declaration
var modifiers: [String] {
    let parser = Formatter(openTokens)

    guard let keywordIndex = parser.index(of: .keyword(keyword), after: 0) else {
        return []
    }

    var allModifiers = [String]()
    _ = parser.modifiersForDeclaration(at: keywordIndex) { _, modifier in
        allModifiers.append(modifier)
        return false
    }

    return allModifiers
}
```

and then:

```swift
let disallowedModifiers = ["override", "@objc", "@IBAction", "@IBOutlet", /* ... */]
let declarationModifiers = Set(declaration.modifiers)
let hasDisallowedModifiers = disallowedModifiers.contains(where: { declarationModifiers.contains($0) })
```

Reply:
Love it - adding it in now